### PR TITLE
feat: add subdoc migration script

### PR DESCRIPTION
**Pull request by Saul-Mirone**

### CODE REVIEW

File: packages/env/src/blocksuite/subdoc-migration.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+export function migrateToSubdoc(doc: Y.Doc): Y.Doc {
+  const needMigration = doc.get('space:meta') instanceof Y.Map;
+  if (!needMigration) {
+    return doc;
+  }
+  const output = new Y.Doc();
+  migrateMeta(doc, output);
+  migrateBlocks(doc, output);
+  return output;
```
Comment (himself65) [inline packages/env/src/blocksuite/subdoc-migration.ts]: Can we use the original doc?

Comment (Saul-Mirone) [reply packages/env/src/blocksuite/subdoc-migration.ts:76]: Seems we can't because we cannot remove a property from a ydoc.

**Review by himself65 — CHANGES_REQUESTED** [review]
I need to add a test case to make sure this works correct

File: packages/env/src/blocksuite/subdoc-migration.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      const version = originalVersions.get(flavour);
+      if (version !== undefined) {
+        runBlockMigration(flavour, blockData, version);
+      }
+    });
+  });
+}
+
+export function migrateToSubdoc(doc: Y.Doc): Y.Doc {
+  const needMigration = doc.get('space:meta') instanceof Y.Map;
```
Comment (himself65) [inline packages/env/src/blocksuite/subdoc-migration.ts]: Wrong logic

**Review by himself65 — CHANGES_REQUESTED** [review]

**Review by himself65 — APPROVED** [review]

**Pull request merged**